### ISSUE TITLE
Footer not fixed to bottom of screen in iOS 4.3

### ISSUE THREAD

**Issue by vincenttheeten** [watusi/jquery-mobile-iscrollview#13]
Thanks for sharing your plugin!

I found your jQuery Mobile iScroll plugin while searching for a truly fixed header and footer bar.

Running your demo in iOS 4.3 shows a footer bar that is not attached to the bottom of the screen. (hovering half way)

In iOS 5+ it works beautifully.

The iScroll4 demo works on iOS 4.3.
I've tried making the footer absolutely positioned (as below).
Doesn't work -- the footer is absolutely positioned, but 20+ pixels off. 

``` css
.ui-footer {position:absolute !important;bottom:0px}
```

**Comment by jtara** [comment]
Thanks for pointing this out. I get somewhat different results - the footer is at the bottom (but currently I'm using a fixed footer) but then there is space above it that isn't used. Seems to have something to do with detection of the browser height, which I get correctly in native apps, but not on Mobile Safari.

It's correct on iPad 1 and 3 (but with ios 5 on both) but also fails on iOS 5.1 on iPhone 4S. So seems to fail for me on iPhone, but OK on iPad. Maybe is related to the Safari navigation bar popping in and out on iPhone.

I assume you are testing in Mobile Safari, could you confirm that?

Frankly, almost all of my testing has been in native apps on devices using a WebView, (as well as desktop browsers), since my personal focus is on native apps.

**Comment by vincenttheeten** [comment]
Yes, I use Mobile Safari on iOS Simulator version 4.3.2

I found http://stackoverflow.com/questions/273671/hiding-safari-user-interface-components-on-iphone
It describes that iOS can hide the Mobile Safari UI **ONLY when launched from the springboard.**

Adding the line below fixes the issue

``` html
<meta name="apple-mobile-web-app-capable" content="yes" />
```

**Comment by jtara** [comment]
I've done a bit of testing on Mobile Safari on iOS Simulator with remote debugging enabled.

Problem is $(window).height() gives inconsistent results. Not sure that there is really any cure. Maybe I can use the height of the HTML element instead, which seems to always report the same value (420px).

The link above is interesting, because the original jquery.mobile.iscroll.js that gave me the idea for this plugin had this "mystery" 44 pixels added to the wrapper if on iOS, and that never made sense to me, so I took it out. It's not needed when using a UIWebView. I don't think the author of that plugin was using a UIWebView in a native app, just the browser.

Now maybe those mystery pixels are explained by the navigation bar. But according to above, they need to be padded to the bottom, not added to the wrapper.

I still have 20 pixel unaccounted for. That sounds like a status bar.

It is possible to detect UIWebView, (absence of "Safari" in user agent, but otherwise iPhone signature), as well as standalone mode (site saved to desktop) with window.navigator.standalone. I probably need to each of these a special case. UIWebView and standalone don't have the issue.

Unfortunately, I may just have to emphasize that the plugin is mainly for native developers using PhoneGap, Titanium, Rhodes, etc. It works well in a UIWebView (as well as desktop browsers). I can't spend much more time on dealing with mobile browsers. Even the JQM demo is pretty flakey on Mobile Safari...

**Comment by jtara** [comment]
If I set wrapperAdd to 64, AND add 64 pixels of padding to the body element, it seems to work correctly.

There is still a problem on orientation change. When I change to landscape, there is space below the footer. But when I switch back to portrait, it is again correct.

So, if I can figure out what is wrong on switch to landscape, I can add some conditional code to detect the various iOS environments and apply the necessary fixes.

**Comment by jtara** [comment]
OK, I think I've got it. At least experimentally. Now, I just have to find out exactly WHY all this is needed, and write the conditional code to apply the fix according to browser environment.

I don't need to add any padding to the page.

I added 64 pixels to the wrapper, using the wrapperAdd option. I can't really explain the extra 20px at this point. Pretty sure the Safari Navigation Bar is 44px...

I added a call to $.mobile.silentscroll(0) at the end of _windowResizeFunc(). This is the function that's called when the browser size or orientation changes (depending on which events are in resizeEvents option).

I only used the resize event, the default.

This works initially and on orientation change.

It fails to change orientation when the screen is portrait upside-down. I think this is a JQuery Mobile issue, though. Haven't tried with 1.1 yet, just working with 1.0.

Oh, I also added a call to $.moble.fixedToolbars.setTouchToggleEnabled(false) just before self-initialization (way at the bottom of jquery.mobile.iscrollview.js).

**Comment by jtara** [comment]
OK, that was almost...

The correct value to add is 60px, not 64px. And now I am pretty sure what it is, though it still baffles me. The address bar that pops down when you navigate is 60px.

It's unclear whether it resizes the window 60px shorter or not, but it seems to scroll it down by that amount in any case.

**Comment by jtara** [comment]
All known problems with incorrect footer positioning should be fixed in 1.2. Last one found was a doozy - jQuery Mobile actually moves persistent footers in and out of the page during transitions, and the widget now tracks that.
